nav2_behavior_tree/plugins/condition/distance_traveled_condition.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019 Intel Corporation
+// Copyright (c) 2020 Sarthak Mittal
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__DISTANCE_TRAVELED_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__DISTANCE_TRAVELED_CONDITION_HPP_
+
+#include <string>
+#include <memory>
+
+#include "nav2_util/robot_utils.hpp"
+#include "nav2_util/geometry_utils.hpp"
+
+#include "distance_traveled_condition.hpp"
+
+namespace nav2_behavior_tree
+{
+
+DistanceTraveledCondition::DistanceTraveledCondition(
```

Comment:
Any other name suggestions? I'm not sure this quite encapsulates it.

Reply:
How about something like `HasTraveled`? As in, has the robot traveled `distance` meters?